File: projects-nodejs/eon/motion.js
var raspi = require('raspi-io');
var five = require('johnny-five');
var board = new five.Board({io: new raspi()});
var PUBNUB = require('pubnub');

var pubnub = PUBNUB.init({
  publish_key: 'demo',
  subscribe_key: 'demo'
});

board.on('ready', function() {

  console.log('board is ready');

  // Create a new `motion` hardware instance.
  var pin = new five.Pin({
    pin: 'P1-7'
  }); //pin 7 (GPIO 4)

  setInterval(function(){

    pin.query(function(state) {

      console.log(state)
      console.log(state.value);
      console.log(state.mode);

      pubnub.publish({
        channel: "c3-gauge-rasp-demo",
        message: {
          columns: [['data', state]]
        }
      });

    });

  }, 1000);

});

File: projects-python/motion-and-range/motion.py
#Accessing the code from other modules
import RPi.GPIO as GPIO
import time
import sys
from pubnub import Pubnub

#Setting up the keys for Pubnub
publish_key = len(sys.argv) > 1 and sys.argv[1] or 'demo'
subscribe_key = len(sys.argv) > 2 and sys.argv[2] or 'demo'
secret_key = len(sys.argv) > 3 and sys.argv[3] or 'demo'
cipher_key = len(sys.argv) > 4 and sys.argv[4] or ''
ssl_on = len(sys.argv) > 5 and bool(sys.argv[5]) or False

## -----------------------------------------------------------------------
## Initiate Pubnub State
## -----------------------------------------------------------------------
pubnub = Pubnub(publish_key=publish_key, subscribe_key=subscribe_key, secret_key=secret_key, cipher_key=cipher_key, ssl_on=ssl_on)
channel = 'MotionDetector'
message = {'motion' : 1}


# Asynchronous usage
def callback(message):
  print(message)

# Setting the pin number as a variable so its easier to use, and setting it to receive an input.
GPIO.setmode(GPIO.BCM)
PIR_PIN = 7
GPIO.setup(PIR_PIN, GPIO.IN)

# calling publish, which sends a message over the channel onto the internet
def MOTION(PIR_PIN):
  pubnub.publish(channel, message, callback=callback, error=callback)

print “PIR Module Test (CTRL+C to exit)”
time.sleep(2)
print “Ready”

# calling the publish only when the sensor detects motion
try:
  GPIO.add_event_detect(PIR_PIN, GPIO.RISING, callback=MOTION)
  while 1:
    time.sleep(100)

except KeyboardInterrupt:
  print “ Quit”
  
GPIO.cleanup()
